tests/integration/__init__.py

```diff
@@ -54,3 +58,30 @@ def set_up_record(self):
                     user_agent=pytest.placeholders.user_agent,
                     refresh_token=pytest.placeholders.refresh_token,
                 )
+
+    def use_cassette(self, cassette_name=None, **kwargs):
+        """Use a cassette. The cassette name is dynamically generated.
+
+        :param cassette_name: (Deprecated) The name to use for the cassette. All names
+            that are not equal to the dynamically generated name will be logged.
+        :param kwargs: All keyword arguments for the main function
+            (``Betamax.use_cassette``).
+        """
+        dynamic_name = self.get_cassette_name()
+        if cassette_name:
+            self.logger.debug(
+                "Static cassette name provided by {}. The following "
+                "name was provided: {}".format(dynamic_name, cassette_name)
+            )
+            if cassette_name != dynamic_name:
+                self.logger.warning(
+                    "Dynamic cassette name for function {} does not "
+                    "match the provided cassette name: {}".format(
+                        dynamic_name, cassette_name
+                    )
+                )
+        return self.recorder.use_cassette(cassette_name or dynamic_name, **kwargs)
+
+    def get_cassette_name(self) -> str:
+        function_name = inspect.currentframe().f_back.f_back.f_code.co_name
+        return "{}.{}".format(type(self).__name__, function_name)
```

Comment:
See above.
```suggestion
        return "{}.{}".format(file_name, function_name)
```

Reply:
This should actually be `class_name`, not `file_name`.

Reply:
That's right, fixed.